### PR TITLE
PT186859156: Integrate & use eslint with prettier

### PR DESCRIPTION
**Pull request by aea-JR**
I would suggest to only look at the first commit ;)
The second commit are the changes after linting the package and removing some warnings

### CODE REVIEW

File: .prettierrc.js [hunk 1]
```diff
@@ -0,0 +1,11 @@
+module.exports = {
+  arrowParens: "avoid",
```
Comment (dawgoc) [inline .prettierrc.js:2]: prettier should automatically detect the eslint and use eslint configuration

File: package.json [hunk 2]
```diff
@@ -9,8 +9,9 @@
     "link": "cd build && npm link",
     "link:vite-project": "npm link",
     "publish:build": "npm run build && cd build && npm publish",
-    "eslint": "eslint --max-warnings 0 src/",
-    "es-check": "npm run build && es-check es5 './build/**/*.js'",
+    "lint": "eslint .",
+    "lint:fix": "eslint --fix .",
+    "prettier": "prettier --write .",
```
Comment (dawgoc) [inline package.json:14]: `"prettier": "prettier --write .",` this shouldn't be necessary

Comment (aea-JR) [reply package.json]: removed it

File: .prettierrc.js (hunk 1 above)
Comment (aea-JR) [reply .prettierrc.js]: removed the config

File: .eslintrc.json [hunk 3]
```diff
@@ -0,0 +1,60 @@
+{
+    "root": true,
+    "parser": "@typescript-eslint/parser",
+    "plugins": [
+        "@typescript-eslint"
```
[the remark this answers is not in the document]
Comment (aea-JR) [reply .eslintrc.json]: Added from document

File: .eslintrc.json [hunk 4]
```diff
@@ -22,11 +22,22 @@
...
+        ]
+      }
     ]
   },
   "extends": [
     "eslint:recommended",
     "plugin:@typescript-eslint/recommended",
+    "plugin:react/recommended",
+    "plugin:react-hooks/recommended",
     "prettier"
```
Comment (dawgoc) [inline .eslintrc.json:41]: `"prettier"` this shouldn't be necessary. I think this actually might overwrite extensions aboce

Comment (aea-JR) [reply .eslintrc.json]: Removed it

**Comment by aea-JR** [comment]
> Have you `npm run lint:fix` after changes?

Yes, i have 2 warnings regarding useEffect but i dont want to address them now as is works as it is.

**Review by dawgoc — APPROVED** [review]

**Comment by aea-JR** [comment]
Thank you ❤️

**Pull request merged**